File: mobile/screens/LoginScreen.js
import { useState } from 'react';
import {
  View,
  Text,
  TextInput,
  Pressable,
  StyleSheet,
  ActivityIndicator,
  KeyboardAvoidingView,
  Platform,
} from 'react-native';
import { useTheme } from '../theme';
import { commonStyles, createThemedStyles, shadows, spacing, typography } from '../styles';
import { login, saveToken } from '../api';

export default function LoginScreen({ onLoggedIn, onGoToRegister, onForgotPassword }) {
  const { colors } = useTheme();
  const themed = createThemedStyles(colors);
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');
  const [error, setError] = useState('');
  const [busy, setBusy] = useState(false);

  async function handleLogin() {
    setError('');
    if (!username.trim() || !password.trim())
      return setError('Please enter your username and password.');
    setBusy(true);
    try {
      const data = await login(username.trim(), password);
      await saveToken(data.token);
      onLoggedIn(data);
    } catch (e) {
      setError(e.message || 'Could not log in. Try again.');
    } finally {
      setBusy(false);
    }
  }

  return (
    <KeyboardAvoidingView
      style={themed.screen}
      behavior={Platform.OS === 'ios' ? 'padding' : undefined}
    >
      <View style={[styles.inner, commonStyles.centered]}>
        <View style={[styles.brandMark, themed.alternateCard, shadows.small]}>
          <Text style={[styles.brandMarkText, themed.primaryText]}>N</Text>
        </View>
        <Text style={[styles.brand, themed.text]}>Noteriety</Text>
        <Text style={[styles.subtitle, themed.mutedText]}>
          A calmer place for your busiest ideas.
        </Text>

        <View style={[commonStyles.card, themed.surfaceCard, shadows.small, styles.formCard]}>
          <Text style={[typography.sectionTitle, themed.text]}>Welcome back</Text>
          <Text style={[typography.bodySmall, themed.mutedText, styles.helper]}>
            Sign in to continue to your notes.
          </Text>
          {error ? <Text style={[styles.error, themed.errorText]}>{error}</Text> : null}

          <Text style={[styles.label, themed.text]}>USERNAME</Text>
          <TextInput
            value={username}
            onChangeText={setUsername}
            autoCapitalize="none"
            autoCorrect={false}
            placeholder="username"
            placeholderTextColor={colors.textMuted}
            style={[commonStyles.input, themed.input]}
          />
          <Text style={[styles.label, themed.text]}>PASSWORD</Text>
          <TextInput
            value={password}
            onChangeText={setPassword}
            secureTextEntry
            autoCapitalize="none"
            placeholder="password"
            placeholderTextColor={colors.textMuted}
            style={[commonStyles.input, themed.input]}
          />

          <Pressable
            onPress={handleLogin}
            disabled={busy}
            style={({ pressed }) => [
              commonStyles.primaryButton,
              themed.primaryButton,
              shadows.button,
              styles.button,
              { opacity: pressed || busy ? 0.75 : 1 },
            ]}
          >
            {busy ? (
              <ActivityIndicator color={colors.onPrimary} />
            ) : (
              <Text style={[typography.button, themed.primaryButtonText]}>Log In</Text>
            )}
          </Pressable>
          <Pressable onPress={onForgotPassword} style={styles.linkButton}>
            <Text style={[styles.link, themed.primaryText]}>Forgot password?</Text>
          </Pressable>
        </View>

        <Pressable onPress={onGoToRegister} style={styles.bottomLink}>
          <Text style={[typography.bodySmall, themed.mutedText]}>
            Don't have an account? <Text style={[styles.link, themed.primaryText]}>Sign up</Text>
          </Text>
        </Pressable>
      </View>
    </KeyboardAvoidingView>
  );
}

const styles = StyleSheet.create({
  inner: { flex: 1, justifyContent: 'center', paddingHorizontal: 22, paddingVertical: 38 },
  brandMark: {
    width: 66,
    height: 66,
    borderWidth: 1,
    borderRadius: 18,
    alignSelf: 'center',
    alignItems: 'center',
    justifyContent: 'center',
  },
  brandMarkText: { fontSize: 31, fontWeight: '800' },
  brand: {
    marginTop: 14,
    fontSize: 34,
    lineHeight: 39,
    fontWeight: '700',
    textAlign: 'center',
    letterSpacing: -0.7,
  },
  subtitle: { marginTop: 7, marginBottom: 28, textAlign: 'center', ...typography.bodySmall },
  formCard: { padding: 22 },
  helper: { marginTop: 6, marginBottom: 10 },
  label: { marginTop: 16, marginBottom: 7, ...typography.label },
  button: { marginTop: 24 },
  linkButton: { alignSelf: 'center', marginTop: 17, padding: 6 },
  bottomLink: { alignSelf: 'center', marginTop: 21, padding: 6 },
  link: { fontWeight: '700', textDecorationLine: 'underline' },
  error: { marginTop: 12, padding: 10, borderRadius: 9, fontSize: 14, textAlign: 'center' },
});

File: noteriety_mobile/lib/screens/login_screen.dart
import 'package:flutter/material.dart';

import '../api/api.dart';
import '../theme/app_colors.dart';
import '../widgets/brand.dart';
import '../widgets/primary_button.dart';

class LoginScreen extends StatefulWidget {
  final VoidCallback onLoggedIn;
  final VoidCallback onGoToRegister;
  final VoidCallback onForgotPassword;

  const LoginScreen({
    super.key,
    required this.onLoggedIn,
    required this.onGoToRegister,
    required this.onForgotPassword,
  });

  @override
  State<LoginScreen> createState() => _LoginScreenState();
}

class _LoginScreenState extends State<LoginScreen> {
  final _username = TextEditingController();
  final _password = TextEditingController();
  String _error = '';
  bool _busy = false;

  @override
  void dispose() {
    _username.dispose();
    _password.dispose();
    super.dispose();
  }

  Future<void> _handleLogin() async {
    setState(() => _error = '');
    final u = _username.text.trim();
    final p = _password.text;
    if (u.isEmpty || p.trim().isEmpty) {
      setState(() => _error = 'Please enter your username and password.');
      return;
    }
    setState(() => _busy = true);
    try {
      final data = await login(u, p);
      final token = data['token'];
      if (token is String && token.isNotEmpty) {
        await saveToken(token);
      }
      if (!mounted) return;
      widget.onLoggedIn();
    } on ApiException catch (e) {
      if (mounted) setState(() => _error = e.message);
    } catch (_) {
      if (mounted) setState(() => _error = 'Could not log in. Try again.');
    } finally {
      if (mounted) setState(() => _busy = false);
    }
  }

  @override
  Widget build(BuildContext context) {
    final colors = context.colors;
    return Scaffold(
      backgroundColor: colors.background,
      body: SafeArea(
        child: Center(
          child: SingleChildScrollView(
            padding: const EdgeInsets.all(28),
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                const Logo(),
                const Eyebrow('WELCOME BACK',
                    padding: EdgeInsets.only(top: 30)),
                const SizedBox(height: 10),
                const Display('Sign in to\nyour notes.', size: 32),
                const SizedBox(height: 22),
                if (_error.isNotEmpty)
                  Padding(
                    padding: const EdgeInsets.only(bottom: 4),
                    child: Text(
                      _error,
                      textAlign: TextAlign.center,
                      style: TextStyle(color: colors.error, fontSize: 14),
                    ),
                  ),
                _label('Username'),
                AppTextField(
                  controller: _username,
                  hint: 'username',
                  capitalization: TextCapitalization.none,
                ),
                _label('Password'),
                AppTextField(
                  controller: _password,
                  hint: 'password',
                  obscure: true,
                  capitalization: TextCapitalization.none,
                ),
                PrimaryButton(
                  label: 'Log In',
                  busy: _busy,
                  onPressed: _handleLogin,
                  margin: const EdgeInsets.only(top: 28),
                ),
                Center(
                  child: TextButton(
                    onPressed: widget.onForgotPassword,
                    child: Text(
                      'Forgot password?',
                      style: TextStyle(color: colors.primary, fontSize: 14),
                    ),
                  ),
                ),
                Center(
                  child: GestureDetector(
                    onTap: widget.onGoToRegister,
                    child: Text.rich(
                      TextSpan(
                        text: "Don't have an account? ",
                        style: TextStyle(color: colors.textMuted, fontSize: 14),
                        children: [
                          TextSpan(
                            text: 'Sign up',
                            style: TextStyle(color: colors.primary),
                          ),
                        ],
                      ),
                    ),
                  ),
                ),
              ],
            ),
          ),
        ),
      ),
    );
  }

  Widget _label(String text) => Padding(
        padding: const EdgeInsets.only(top: 14, bottom: 6),
        child: Text(
          text,
          style: TextStyle(
            fontSize: 13,
            fontWeight: FontWeight.w600,
            color: context.colors.text,
          ),
        ),
      );
}
